src/libs/meta/meta.c

```diff
@@ -989,6 +1002,14 @@ static int topCmpOrder (const void * a, const void * b)
 	return strcmp (keyString (kam), keyString (kbm));
 }
 
+/*
+ * does topological sorting on "ks" using the dependencies defined by the "dep/#" metakeys
```

Comment:
solving transitive dependencies is pretty much the core functionality of every topological sort algorithm, not sure what to say about that part.